nifi-commons/nifi-sensitive-property-provider/src/test/java/org/apache/nifi/properties/GCPSensitivePropertyProviderIT.java

```diff
@@ -0,0 +1,125 @@
+import org.mockito.internal.util.io.IOUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Properties;
+
+/**
+ * To run this test, make sure to first configure sensitive credential information as in the following link
+ * https://cloud.google.com/kms/docs/reference/libraries#cloud-console
+ *
+ * Create a project, keyring and key in the web console.
+ *
+ * Take note of the project name, location, keyring name and key name.
+ *
+ * Then, set the system properties as follows:
+ * -Dgcp.kms.project="project"
+ * -Dgcp.kms.location="location"
+ * -Dgcp.kms.keyring="key ring name"
+ * -Dgcp.kms.key="key name"
+ * when running the integration tests
+ */
+
+public class GCPSensitivePropertyProviderIT {
+    private static final String SAMPLE_PLAINTEXT = "GCPSensitivePropertyProviderIT SAMPLE-PLAINTEXT";
+    private static final String PROJECT_ID_PROPS_NAME = "gcp.kms.project";
+    private static final String LOCATION_ID_PROPS_NAME = "gcp.kms.location";
+    private static final String KEYRING_ID_PROPS_NAME = "gcp.kms.keyring";
+    private static final String KEY_ID_PROPS_NAME = "gcp.kms.key";
+    private static final String BOOTSTRAP_GCP_FILE_PROPS_NAME = "nifi.bootstrap.protection.gcp.kms.conf";
+
+    private static final String EMPTY_PROPERTY = "";
+
+    private static GCPSensitivePropertyProvider spp;
+
+    private static BootstrapProperties props;
+
+    private static Path mockBootstrapConf, mockGCPBootstrapConf;
+
+    private static final Logger logger = LoggerFactory.getLogger(GCPSensitivePropertyProviderIT.class);
+
+    private static void initializeBootstrapProperties() throws IOException{
+        mockBootstrapConf = Files.createTempFile("bootstrap", ".conf").toAbsolutePath();
+        mockGCPBootstrapConf = Files.createTempFile("bootstrap-gcp", ".conf").toAbsolutePath();
+        IOUtil.writeText(BOOTSTRAP_GCP_FILE_PROPS_NAME + "=" + mockGCPBootstrapConf.toAbsolutePath(), mockBootstrapConf.toFile());
+
+        final Properties bootstrapProperties = new Properties();
+        try (final InputStream inputStream = Files.newInputStream(mockBootstrapConf)) {
+            bootstrapProperties.load(inputStream);
+            props = new BootstrapProperties("nifi", bootstrapProperties, mockBootstrapConf);
+        }
+
+        String projectId = System.getProperty(PROJECT_ID_PROPS_NAME, EMPTY_PROPERTY);
+        String locationId = System.getProperty(LOCATION_ID_PROPS_NAME, EMPTY_PROPERTY);
+        String keyringId = System.getProperty(KEYRING_ID_PROPS_NAME, EMPTY_PROPERTY);
+        String keyId = System.getProperty(KEY_ID_PROPS_NAME, EMPTY_PROPERTY);
+
+        StringBuilder bootstrapConfText = new StringBuilder();
+        bootstrapConfText.append(PROJECT_ID_PROPS_NAME + "=" + projectId);
+        bootstrapConfText.append("\n" + LOCATION_ID_PROPS_NAME + "=" + locationId);
```

Comment:
Would it be better to use System.getProperty("line.separator") here?

Reply:
I think that would be better yes, I will make the changes with the AWS and GCP SPP ITs!